pubsub/unit_tests/test__gax.py

```diff
@@ -935,13 +937,15 @@ def test_live_api(self):
         from google.cloud.pubsub._gax import DEFAULT_USER_AGENT
 
         channels = []
+        publisher_api_kwargs = {}
         channel_args = []
         channel_obj = object()
         mock_result = object()
         host = 'foo.apis.invalid'
 
-        def mock_publisher_api(channel):
+        def mock_publisher_api(channel, **kwargs):
             channels.append(channel)
+            publisher_api_kwargs.update(kwargs)
```

Comment:
This is "lossy", i.e. it can get clobbered on multiple calls. Instead `publisher_api_kwargs.append(kwargs)` (so make it a list).

Or you could do the dirty work and use a real mock?